File: Leetcode/Python/_1814.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        def rev(n):
            acc = 0
            while n:
                acc = acc * 10 + n % 10
                n //= 10

            return acc

        arr = []
        for i in range(len(nums)):
            arr.append(nums[i] - rev(nums[i]))

        MOD = 10**9 + 7
        mapping = defaultdict(int)
        output = 0
        for i in range(len(nums)):
            output = (output + mapping[arr[i]]) % MOD
            mapping[arr[i]] += 1
        return output

File: Leetcode/cpp/1814.cpp
class Solution {
public:
    int rev(int n) {
        int acc = 0;
        while (n > 0) {
            acc = acc * 10 + n % 10;
            n /= 10;
        }
        return acc;
    } 
    int countNicePairs(vector<int>& nums) {
        vector<int> arr;
        unordered_map<int, int> dict;
        int output = 0, MOD = 1e9 + 7;
        for (int i = 0; i < nums.size(); i++) {
            arr.push_back(nums[i] - rev(nums[i]));
        }
        for (int num : arr) {
            output = (output + dict[num]) % MOD;
            dict[num]++;
        }
        return output;
    }
};
